release-notes/Cabal-3.10.2.0.md

```diff
@@ -0,0 +1,32 @@
+Cabal and Cabal-syntax 3.10.2.0 changelog and release notes
+---
+
+- Shorten script-builds paths [#8841](https://github.com/haskell/cabal/issues/8841) [#8898](https://github.com/haskell/cabal/pull/8898)
+  
+  - Use Base64 hash truncated to 26 chars for script-build cache directories.
+  - Use the cache directory as the dist directory.
+  - Use script-<your-sanitized-script-name> as the component name instead of cabal-script-<...>.
+  - Use cabal-script-<your-actual-script-name> for the executable name.
+  - This change is incompatible with previous cabal versions in terms of cache location,
+    you should manually remove your old caches once you no longer need them.
+  
+- Do not always pass --quickjump to haddock #9049 [#9060](https://github.com/haskell/cabal/issues/9060) [#9049](https://github.com/haskell/cabal/pull/9049)
+  
+  6d8adf13101 caused `cabal` to always pass the `--quickjump` flag to Haddock.
+  Not only does this waste memory for a service that user hasn't asked for,
+  but also leads to a failure with Haddocks shipped with GHC 9.0 and 9.2,
+  which had a separate bug (fixed in later versions but not backported) when
+  Haddock does not pass `--quickjump` recursively to the package dependencies.
+  
+### Significant changes
```

Comment:
I guess, a part of the manual editing is categorizing the changes above as significant or not.